app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class PaymentsController < ManagerController
+  before_action :find_project
+  before_action :find_payment, only: %i[destroy update]
+
+  def create
+    @payment = @project.payments.new(payment_params)
+
+    if @payment.save
+      flash[:notice] = 'successfully added payment'
+      redirect_to project_path(@project)
+    else
+      @project.payments.delete(@payment)
```

Comment:
i noticed this in some controller above too, why are we deleting and what mechanism are we using for showing error on frontend?